roles/grafana_cloud_operator/tasks/delete_grafana_oncall_hub_spoke.yml

```diff
@@ -0,0 +1,156 @@
+        - lastTransitionTime: "{{ ansible_date_time.iso8601 }}"
+          status: "True"
+          type: "DeletionInitiated"
+          reason: "OperationStarted"
+          message: "Operation to delete Grafana OnCall integration for hubAndSpoke started"
+
+- name: Fetch ManagedCluster from all namespaces
+  kubernetes.core.k8s_info:
+    api_version: cluster.open-cluster-management.io/v1
+    kind: ManagedCluster
+  register: managed_clusters_raw
+
+- name: Extract list of ManagedCluster
+  ansible.builtin.set_fact:
+    managed_clusters: "{{ managed_clusters | default([]) + [{'name': item.metadata.name}] }}"
+  loop: "{{ managed_clusters_raw.resources }}"
+  loop_control:
+    loop_var: item
+
+- name: Determine which ManagedCluster don't have integrations
+  ansible.builtin.set_fact:
+    create_integration_for: "{{ managed_clusters | rejectattr('name', 'in', existing_integration_names) | list }}"
+
+- name: Fetch current status of Config CR
+  kubernetes.core.k8s_info:
+    api_version: grafanacloud.stakater.com/v1alpha1
+    kind: Config
+    name: "{{ cr_name }}"
+    namespace: "{{ cr_namespace }}"
+  register: config_cr
+
+- name: Set current managed clusters from status
+  ansible.builtin.set_fact:
+    previous_managed_clusters: "{{ config_cr.resources[0].status.managedClusters | default([]) }}"
+
+- name: Extract current managed cluster names
+  ansible.builtin.set_fact:
+    current_managed_clusters: "{{ managed_clusters | map(attribute='name') | list }}"
+
+- name: Determine integrations to delete in Grafana Cloud
+  ansible.builtin.set_fact:
+    delete_integration_for: "{{ previous_managed_clusters | difference(current_managed_clusters) }}"
+
+- name: Print delete_integration_for
+  ansible.builtin.debug:
+    var: delete_integration_for
+
+- name: Extract integration details from the response
+  ansible.builtin.set_fact:
+    grafana_integrations: "{{ existing_integrations_response.json.results }}"
+
+- name: Find Grafana Integration IDs for the Integrations to Delete
+  ansible.builtin.set_fact:
+    integrations_to_delete: "{{ integrations_to_delete | default([]) + [{'name': item, 'integration_id': integration_id}] }}"
+  loop: "{{ delete_integration_for }}"
+  vars:
+    integration_id: >-
+      {{
+        grafana_integrations | selectattr('name', 'eq', item) | map(attribute='id') | first | default(None)
+      }}
+  when: integration_id is not none
+
+- name: Delete integrations from Grafana Cloud for deleted ManagedClusters
+  ansible.builtin.uri:
+    url: "{{ grafana_cloud_operator_grafana_cloud_integrations_api_url }}{{ item.integration_id }}"
+    method: DELETE
+    headers:
+      Authorization: "{{ grafana_cloud_api_token }}"
+      Content-Type: "application/json"
+    status_code: [200, 201]
+  loop: "{{ integrations_to_delete }}"
+  loop_control:
+    label: "{{ item }}"
+  retries: 5
+  delay: 6
+  when: delete_integration_for
+  failed_when: false
+
+- name: Update CR status for Integrations Deleted
```

Comment:
Update CR status for Integrations Deleted => Update CR status for deleted Integrations